### PR TITLE
Bring Int63 notations into line with stdlib

### LINKED ISSUE #12454: Coq Standard Library should be consistent about using "<" and "<=" for prop-valued things and "<?" and "<=?" for bool-valued things
#### Description of the problem

https://github.com/coq/coq/blob/0ae20176df146b90f11e131aff3673584e6ffb3a/theories/Numbers/Cyclic/Int63/Int63.v#L72-L80
This is inconsistent with the notations for most other types.  The standard is to use `=?` for boolean equality, `<?` for ltb, and `<=?` for `leb`.

cc @maximedenes 

Is fixing this a candidate for 8.12?

#### Coq Version

master

### PR DESCRIPTION
**Pull request by JasonGross**
We also put them in a module, so users can `Require Int63. Import
Int63.Int63Notations` without needing to unqualify the primitives.

In particular, we change
- `a \% m` into `a mod m` to correspond with the notation in ZArith
- `m == n` into `m =? n` to correspond with the eqb notations elsewhere
- `m < n` into `m <? n` to correspond with the ltb notations elsewhere
- `m <= n` into `m <=? n` to correspond with the leb notations elsewhere
- `m ≤ n` into `m ≤? n` for consistency with the non-unicode notation

The old notations are still accessible as deprecated notations.

**Kind:** bug fix

Fixes / closes #12454

- [x] Added / updated test-suite
- Corresponding documentation was added / updated (including any warning and error messages added / removed / modified). (doesn't seem to be any documentation to update)
- [x] Entry added in the changelog (see https://github.com/coq/coq/tree/master/doc/changelog#unreleased-changelog for details).

Is this a candidate for 8.12?

### CODE REVIEW

**Comment by ejgallego** [comment]
> Is this a candidate for 8.12?

@Zimmi48 , what do you think? From my part if the stdlib maintainers approve of this I'm OK with having this in 8.12.

**Comment by ejgallego** [comment]
Something @coq/stdlib-maintainers to take into consideration is that if you think this could be a 8.12 candidate, it should be ideally shipped in the beta so please have a look soon.

**Comment by Zimmi48** [comment]
Agreed if it gets approved and merged in time for the beta.

**Comment by proux01** [comment]
@JasonGross note that the same might be needed for primitive floats as the notations for boolean comparisons where copied from Int63.

**Review by maximedenes — APPROVED** [review]
No strong opinion, but LGTM if it makes int63 notations closer to the rest of stdlib.

**Comment by JasonGross** [comment]
I guess I need to update this to say that the notations are deprecated since 8.13, now, since it didn't make it into 8.12?

**Comment by erikmd** [comment]
BTW I was wondering if this PR (which happens to replace the `≤` infix with a non-Unicode notation `<=?`) could be the opportunity to rephrase these two notations as well, to non-Unicode? (or at least, put them `only parsing`)

```
Notation "'φ' x" := (to_Z x) (at level 0) : int63_scope.
Notation "'Φ' x" :=
   (zn2z_to_Z wB to_Z x) (at level 0) : int63_scope. 
```
because I believe that Unicode notations are scarce otherwise in the stdlib, and it can be difficult to input them otherwise than by copying-pasting (if the users relying on the `Int63` library don't have some dedicated input methods in their editor nor the availability of Linux' compose key, for example…)

WDYT?

**Comment by JasonGross** [comment]
> (which happens to replace the `≤` infix with a non-Unicode notation `<=?`)

It doesn't, actually.  It replaces `<=` with `<=?` and `≤` with `≤?`.

> could be the opportunity to rephrase these two notations as well, to non-Unicode? (or at least, put them `only parsing`)

I would support this change, but don't feel strongly about it.  I'd rather leave it to another PR; I see the changes in this PR as not-really-subject-to-bikeshedding (there's a clear convention in the rest of the stdlib, and an obvious way to bring these notations in line), whereas it's not clear to me what the non-unicode version should be for these notations.  And the `<=` vs `<=?` seems more likely to cause confusion and/or notation conflicts, whereas changing the morphism symbol seems much more amenable to find-and-replace fixing.

**Comment by JasonGross** [comment]
Who should be assignee?  @maximedenes ?

**Review by herbelin — APPROVED** [review]

**Comment by JasonGross** [comment]
I've rebased this.  Can someone please assign and merge (assuming it hasn't already bitrotted)?

**Comment by anton-trunov** [comment]
Ok, let me self-assign if nobody minds. The PR looks good to me and CI is green. I'll merge it on Thursday unless someone objects.

**Pull request merged**
